File: backend/src/ERPBPM.Domain/Entities/Position.cs
using ERPBPM.Domain.Common;
using ERPBPM.Domain.Enums;

namespace ERPBPM.Domain.Entities;

public class Position : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string Code { get; set; } = string.Empty;
    public string? Description { get; set; }
    public PositionLevel Level { get; set; }
    public decimal MinSalary { get; set; }
    public decimal MaxSalary { get; set; }
    public bool IsActive { get; set; } = true;

    // Navigation properties
    public ICollection<Employee> Employees { get; set; } = new List<Employee>();
}

File: frontend-mobile/lib/features/hrm/domain/entities/position.dart
import 'package:equatable/equatable.dart';

class Position extends Equatable {
  final String id;
  final String code;
  final String title;
  final String? description;
  final String level;
  final double minSalary;
  final double maxSalary;
  final int employeeCount;
  final bool isActive;

  const Position({
    required this.id,
    required this.code,
    required this.title,
    this.description,
    required this.level,
    required this.minSalary,
    required this.maxSalary,
    required this.employeeCount,
    required this.isActive,
  });

  @override
  List<Object?> get props => [
        id,
        code,
        title,
        description,
        level,
        minSalary,
        maxSalary,
        employeeCount,
        isActive,
      ];
}
